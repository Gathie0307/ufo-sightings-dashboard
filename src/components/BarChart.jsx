import {
  Chart as ChartJS,
  BarElement,
  CategoryScale,
  LinearScale,
  Tooltip,
  Legend,
  Title,
} from 'chart.js';
import { Bar } from 'react-chartjs-2';

ChartJS.register(
  BarElement,
  CategoryScale,
  LinearScale,
  Tooltip,
  Legend,
  Title
);

const BarChart = ({ data, options }) => {
  return <Bar data={data} options={options}></Bar>;
};

export default BarChart;
