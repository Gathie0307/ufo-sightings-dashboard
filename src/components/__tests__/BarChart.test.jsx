import { render } from '@testing-library/react';
import BarChart from '../BarChart';

describe('Bar Chart', () => {
  beforeAll(() => {
    Object.defineProperty(HTMLCanvasElement.prototype, 'getContext', {
      value: vi.fn(() => {}),
    });
  });

  it('should match snapshot', () => {
    const data = {
      labels: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'],
      datasets: [
        {
          label: 'Sightings',
          data: [23, 11, 50, 34, 25, 1, 20],
          backgroundColor: [
            'rgba(255, 26, 104, 0.2)',
            'rgba(54, 162, 235, 0.2)',
          ],
          barThickness: 50,
          borderRadius: 5,
        },
      ],
    };
    const options = {
      plugins: {
        title: {
          display: true,
          text: 'Weekly UFO Sightings',
        },
        legend: {
          display: true,
          position: 'bottom',
        },
      },
    };
    const { asFragment } = render(<BarChart data={data} options={options} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
