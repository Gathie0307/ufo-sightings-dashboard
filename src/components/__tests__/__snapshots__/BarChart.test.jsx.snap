// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bar Chart > should match snapshot 1`] = `
<DocumentFragment>
  <canvas
    height="150"
    role="img"
    width="300"
  />
</DocumentFragment>
`;
